### PR TITLE
[feature] Better squash to tk1

### PR DESCRIPTION
**Pull request by lmondada**

### CODE REVIEW

File: tket/tests/test_Synthesis.cpp [hunk 1]
```diff
@@ -2186,5 +2186,124 @@ SCENARIO("Restricting ZZPhase gate angles.") {
...
+    circ.add_op<unsigned>(OpType::T, {1});
+    circ.add_op<unsigned>(OpType::Sdg, {1});
+    circ.add_op<unsigned>(OpType::Rz, 2.434, {0});
+    circ.add_op<unsigned>(OpType::CZ, {0, 1});
+    circ.add_op<unsigned>(OpType::Rx, 22.22, {1});
+    circ.add_op<unsigned>(OpType::Rx, 3.334, {0});
+  }
+
+  auto u0 = tket_sim::get_unitary(circ);
+  Transforms::decompose_ZXZ_to_TK1().apply(circ);
```
Comment (yao-cqc) [inline tket/tests/test_Synthesis.cpp:2243]: Maybe require this to return true and also test that the rebased circuits only have tk1 and 2-q gates? Similar for the ZYZ test below.

**Review by yao-cqc — CHANGES_REQUESTED** [review]
Can you add a PR comment explaining why this improves tk1 squash?

File: tket/tests/test_Synthesis.cpp (hunk 1 above)
Comment (lmondada) [reply tket/tests/test_Synthesis.cpp]: Done.

**Comment by lmondada** [comment]
I've added a comment in the code about why this rebase sequence makes sense. Apologies for all the trashy commits.

**Review by yao-cqc — APPROVED** [review]

**Pull request merged**